netbox/utilities/versions.py

```diff
@@ -0,0 +1,27 @@
+import requests
+from cacheops import cached
+from django.conf import settings
+from packaging import version
+
+if settings.GITHUB_VERSION_TIMEOUT and settings.GITHUB_REPOSITORY:
+    @cached(timeout=settings.GITHUB_VERSION_TIMEOUT)
+    def get_latest_version():
+        url = 'https://api.github.com/repos/{}/releases'.format(settings.GITHUB_REPOSITORY)
+        headers = {
+            'Accept': 'application/vnd.github.v3+json',
+        }
+        try:
+            response = requests.get(url, headers=headers)
+            versions = [(version.parse(release['tag_name']), release.get('html_url'))
+                        for release in response.json()
+                        if 'tag_name' in release]
+            if versions:
+                return max(versions)
+        except Exception:
+            pass
+
+        return 'unknown', None
+
+else:
+    def get_latest_version():
+        return None
```

Comment:
This will throw a `cannot unpack non-iterable NoneType object` error when it is called.
```python
github_latest_version, github_url = get_latest_version()
```

Maybe move the condition to inside the function 

```python
def get_latest_version():
    if settings.GITHUB_VERSION_TIMEOUT and settings.GITHUB_REPOSITORY:
        # check the version and attempt to return the newest one
        ...

    return 'unknown', None
```

Reply:
The reason I put the `if` outside the function is that using the decorator with `timeout=0` will raise an exception. I also don't want to cache whether the setting is turned on/off as that would confuse users.

As the added complexity is caused by using a decorator I'll rewrite it to use the `cache` object directly. This makes the code a lot more readable.

Reply:
I switched to another solution: split the function into two: `get_releases`, which uses the decorator with a minimum `timeout=1` and `get_latest_release` which is used by the context processor.